core/src/main/java/hudson/PluginManager.java

```diff
@@ -1643,26 +1649,73 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
     @Restricted(NoExternalUse.class)
     @RequirePOST public HttpResponse doCheckUpdatesServer() throws IOException {
         Jenkins.getInstance().checkPermission(Jenkins.ADMINISTER);
-        try {
-            for (UpdateSite site : Jenkins.getInstance().getUpdateCenter().getSites()) {
-                FormValidation v = site.updateDirectlyNow(DownloadService.signatureCheck);
-                if (v.kind != FormValidation.Kind.OK) {
-                    // TODO crude but enough for now
-                    return v;
-                }
+
+        int sleep = SystemProperties.getInteger(PluginManager.class.getName() + ".sleepTimeMilis", 1000);
+        int retries = SystemProperties.getInteger(PluginManager.class.getName() + ".retries", 3);
+
+        retries = retries > 0 ? retries : 3;
+        sleep = sleep > 0 ? sleep : 1000;
+
+        FormValidation response = null;
+        int i = 0;
+        boolean success = false;
+        while(i < retries && !success) {
```

Comment:
Can't you combine the two variables into a for loop?

Reply:
As the two variables have different types and cannot be defined in the initialization section of the for bucle, I prefer to leave it so